backend/src/main/java/xbdd/resources/Report.java

```diff
@@ -0,0 +1,259 @@
+				.sort(Coordinates.getFeatureSortingObject());
+		final BasicDBList featureIndex = new BasicDBList();
+		try {
+			for (final Object o : features) {
+				featureIndex.add(o);
+			}
+		} finally {
+			features.close();
+		}
+		return Response.ok(SerializerUtil.serialise(featureIndex)).build();
+	}
+
+	protected void embedTestingTips(final BasicDBList featureList, final Coordinates coordinates, final DB db) {
+		for (final Object feature : featureList) {
+			Feature.embedTestingTips((DBObject) feature, coordinates, db);
+		}
+	}
+
+	@GET
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("/tags/{product}/{major}.{minor}.{servicePack}/{build}")
+	public Response getTagList(@BeanParam final Coordinates coordinates) {
+		final DB bdd = this.client.getDB("bdd");
+		final DBCollection features = bdd.getCollection("features");
+		List<BasicDBObject> objectList = new ArrayList<>();
+		// Build objects for aggregation pipeline
+		// id option: returns each tag with a list of associated feature ids
+		objectList.add(new BasicDBObject("$match", coordinates.getReportCoordinatesQueryObject()));
+		final DBObject fields = new BasicDBObject("tags.name", 1);
+		fields.put("_id", 0); // comment out for id option
+		objectList.add(new BasicDBObject("$project", fields));
+		objectList.add(new BasicDBObject("$unwind", "$tags"));
+		final DBObject groupFields = new BasicDBObject("_id", "$tags.name");
+		// groupFields.put("features", new BasicDBObject("$addToSet", "$_id")); //comment in for id option
+		groupFields.put("amount", new BasicDBObject("$sum", 1));
+		objectList.add(new BasicDBObject("$group", groupFields));
+		objectList.add(new BasicDBObject("$sort", new BasicDBObject("amount", -1)));
+
+		AggregationOptions options = AggregationOptions.builder().build();
+
+		final Cursor output = features.aggregate(objectList, options);
+
+		// get _ids from each entry of output.result
+		final BasicDBList returns = new BasicDBList();
+		while (output.hasNext()) {
+			returns.add(output.next().get("_id").toString());
+		}
+		return Response.ok(SerializerUtil.serialise(returns)).build();
+	}
+
+	@PUT
+	@Path("/{product}/{major}.{minor}.{servicePack}/{build}")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response putReport(@BeanParam final Coordinates coordinates, final List<JUnitFeature> root) {
+		final MongoDatabase bdd = this.client.getDatabase("bdd");
+
+		final FeatureMapper featureMapper = new FeatureMapper(this.imageDao);
+		this.summaryDao.updateSummary(coordinates);
+
+		List<XbddFeature> mappedFeatures = root.stream().map(feature -> featureMapper.map(feature, coordinates)).collect(Collectors.toList());
+		this.featureDao.saveFeatures(mappedFeatures);
+
+		List<XbddFeature> persistedFeatures = this.featureDao.getFeatures(coordinates);
+
+		this.statsDao.updateStatsForFeatures(coordinates, persistedFeatures);
+
+		return Response.ok(SerializerUtil.serialise(persistedFeatures)).build();
```

Comment:
I don't think you need to call serialise anymore when you've got proper DTOs to return - should be able to just return the DTOs directly and let it serialise itself.

Reply:
done